File: sol/solution/0001-0100/0011. Container With Most Water/Solution.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var left = 0, right = height.count - 1
        var maxArea = 0
        while left < right {
            let width = right - left
            let currentHeight = min(height[left], height[right])
            maxArea = max(maxArea, width * currentHeight)
            if height[left] < height[right] {
                left += 1
            } else {
                right -= 1
            }
        }
        return maxArea
    }
}

File: sol/solution/0001-0100/0011. Container With Most Water/Solution.go
func maxArea(height []int) int {
    left, right := 0, len(height)-1
    maxArea := 0
    for left < right {
        width := right - left
        currentHeight := height[left]
        if height[right] < currentHeight {
            currentHeight = height[right]
        }
        area := width * currentHeight
        if area > maxArea {
            maxArea = area
        }
        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }
    return maxArea
}